src/pluto/backup/BackupManager.ts

```diff
@@ -37,21 +38,24 @@ export class BackupManager {
     await task.run();
   }
 
-  private getBackupTask(version: string = Domain.Backup.defaultVersion): IBackupTask {
+  private getBackupTask(version: Version = Domain.Backup.defaultVersion): IBackupTask {
     switch (version) {
       case "0.0.1":
         return new Versions.v0_0_1.BackupTask(this.Pluto, this.Repositories);
+      case "0.0.2":
+        return new Versions.v0_0_2.BackupTask(this.Pluto, this.Repositories);
     }
 
     throw new Domain.PlutoError.BackupNotFoundError();
   }
 
   private getRestoreTask(backup: Domain.Backup.Schema): IRestoreTask {
-    const version = backup.version ?? Domain.Backup.defaultVersion;
-
-    switch (version) {
+    backup.version = backup.version ?? Domain.Backup.defaultVersion;
+    switch (backup.version) {
```

Comment:
can't say I understand the reason for this change to mutating the backup object?

Reply:
I really don't remember reason but go consistency of backup objects I felt like it may good to have actual version of schema if it was not even passed by backup data